rasa/core/policies/ted_policy.py

```diff
@@ -783,78 +878,62 @@ def persist(self, path: Union[Text, Path]) -> None:
             shutil.move(self.tmp_checkpoint_dir, model_path / "checkpoints")
         self.model.save(str(tf_model_file))
 
+        self.persist_model_utilities(model_path)
+
+    def persist_model_utilities(self, model_path: Path) -> None:
+        """Persist model's utility attributes like model weights, etc.
+
+        Args:
+            model_path: Path where model is to be persisted
+        """
+        metadata_filename = self._metadata_filename()
         io_utils.json_pickle(
-            model_path / f"{SAVE_MODEL_FILE_NAME}.priority.pkl", self.priority
+            model_path / f"{metadata_filename}.priority.pkl", self.priority
         )
+        io_utils.pickle_dump(model_path / f"{metadata_filename}.meta.pkl", self.config)
         io_utils.pickle_dump(
-            model_path / f"{SAVE_MODEL_FILE_NAME}.meta.pkl", self.config
+            model_path / f"{metadata_filename}.data_example.pkl", self.data_example,
```

Comment:
Similar to my comment on IntentTED, we could make these private properties of the class, so it's ensured that they are exactly the same as in the load methods.